homeassistant/components/seventeentrack/__init__.py

```diff
@@ -1 +1,194 @@
+    """Get the latest data from SeventeenTrack."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: ConfigEntry,
+        client: SeventeenTrackClient,
+    ) -> None:
+        """Initialize the data object."""
+        self.hass = hass
+        self.config_entry: ConfigEntry = config_entry
+        self.client = client
+        self.account_id: str = client.profile.account_id
+        super().__init__(
+            self.hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(
+                minutes=self.config_entry.options.get(
+                    CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+                )
+            ),
+        )
+
+    @property
+    def show_archived(self) -> bool:
+        """Include archived packages when fetching data."""
+        return self.config_entry.options.get(CONF_SHOW_ARCHIVED, DEFAULT_SHOW_ARCHIVED)
+
+    async def _async_update_data(self) -> dict[str, dict]:
+        """Update SeventeenTrack data."""
+        try:
+            packages = await self.client.profile.packages(
+                show_archived=False, tz=str(self.hass.config.time_zone)
+            )
+            summary = await self.client.profile.summary(show_archived=False)
+
+            if self.show_archived:
+                archived_packages = await self.client.profile.packages(
+                    show_archived=True, tz=str(self.hass.config.time_zone)
+                )
+                packages += archived_packages
+
+                archived_summary = await self.client.profile.summary(show_archived=True)
+                for status, qty in archived_summary.items():
+                    summary[status] += qty
+
+        except SeventeenTrackError as err:
+            raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+        _LOGGER.debug("New package data received: %s", packages)
+        _LOGGER.debug("New summary data received: %s", summary)
+
+        return {"packages": packages, "summary": summary}
+
+    async def async_register_service(self) -> bool:
+        """Set up SeventeenTrack."""
+
+        async def async_add_package(service: ServiceCall) -> None:
+            """Add new package."""
+            device_registry: DeviceRegistry = (
+                self.hass.helpers.device_registry.async_get(self.hass)
+            )
+            device = device_registry.async_get(service.data[CONF_DEVICE_ID])
+            if device is None:
+                _LOGGER.error("17Track device not found")
+                return
+            for config_entry_id in device.config_entries:
+                config_entry = self.hass.config_entries.async_get_entry(config_entry_id)
+                if config_entry and config_entry.domain == DOMAIN:
+                    client: SeventeenTrackClient = self.hass.data[DOMAIN][
+                        config_entry_id
+                    ].client
+                    break
```

Comment:
The code could lead to a NameError below.

Maybe it's easier to ask for the config entry id in the service call data using that selector? We support that nowadays.